CefSharp.Wpf/IME/IMEHandler.cs

```diff
@@ -0,0 +1,201 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using CefSharp.Structs;
+using static CefSharp.Wpf.IME.NativeIME;
+
+namespace CefSharp.Wpf.IME
+{
+    public class IMEHandler : IDisposable
+    {
+        internal const uint ColorUNDERLINE = 0xFFFFFFFF;   // Black SkColor value for underline,
+        internal const uint ColorBKCOLOR = 0xFF000000;    // White SkColor value for background,
+
+        IntPtr _hWnd;
+
+        internal static IMEHandler Create(IntPtr hWnd)
+        {
+            return new IMEHandler(hWnd);
+        }
+
+        private IMEHandler(IntPtr hWnd)
+        {
+            _hWnd = hWnd;
+        }
+
+        public void Dispose()
+        {
+        }
+
+        private bool GetString(IntPtr hIMC, uint lParam, uint type, out string text)
+        {
+            text = string.Empty;
+
+            if (!IsParam(lParam, type))
+                return false;
+
+            var strLen = ImmGetCompositionString(hIMC, type, null, 0);
+            if (strLen <= 0)
+                return false;
+
+            // buffer contains char (2 bytes)
+            byte[] buffer = new byte[strLen];
+            ImmGetCompositionString(hIMC, type, buffer, strLen);
+            text = Encoding.Unicode.GetString(buffer);
+
+            return true;
+        }
+
+        internal bool GetResult(uint lParam, out string text)
+        {
+            IntPtr hIMC = ImmGetContext(_hWnd);
+
+            var ret = GetString(hIMC, lParam, GCS_RESULTSTR, out text);
+
+            ImmReleaseContext(_hWnd, hIMC);
+
+            return ret;
+        }
+
+        internal bool GetComposition(uint lParam, List<CompositionUnderline> underlines, ref int compositionStart, out string text)
+        {
+            IntPtr hIMC = ImmGetContext(_hWnd);
+
+            bool ret = GetString(hIMC, lParam, GCS_COMPSTR, out text);
+            if (ret)
+			{
+                GetCompositionInfo(lParam, text, underlines, ref compositionStart);
+			}
```

Comment:
nit: I think the CefSharp project uses spaces and not tabs. Please remove the tabs.